### PR TITLE
Create rds-monitoring chart

### PR DESCRIPTION
**Pull request by iuriaranda**
This basically is a wrapper chart to set some default values for cloudwatch-exporter.
It's good to go, the only needed value is the IAM role ARN to use.

As per https://github.com/skyscrapers/engineering/issues/197

### CODE REVIEW

File: rds-monitoring/values.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  serviceAccount:
+    create: true
+
+  rbac:
+    create: true
+
+  config: |-
+    region: eu-west-1
+    period_seconds: 60
```
Comment (simonrondelez) [inline rds-monitoring/values.yaml:45]: this interval might become costly -> to evaluate

File: rds-monitoring/values.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        aws_dimensions: [DBInstanceIdentifier]
+        aws_statistics: [Minimum, Maximum, Average, Sum]
+      - aws_namespace: AWS/RDS
+        aws_metric_name: ReadIOPS
+        aws_dimensions: [DBInstanceIdentifier]
+        aws_statistics: [Minimum, Maximum, Average, Sum]
+      - aws_namespace: AWS/RDS
+        aws_metric_name: WriteIOPS
+        aws_dimensions: [DBInstanceIdentifier]
+        aws_statistics: [Minimum, Maximum, Average, Sum]
```
Comment (simonrondelez) [inline rds-monitoring/values.yaml]: i'm missing replica lag

i'm also not sure if all of these are compatible with Aurora

**Review by duboisph — DISMISSED** [review]

File: rds-monitoring/values.yaml (hunk 1 above)
Comment (duboisph) [reply rds-monitoring/values.yaml:45]: It's not the exporter's period_seconds that makes it costly, but the ServiceMonitor's scraping `interval` higher up.

Still valid comment though, as that's also set to 60 😅

File: rds-monitoring/values.yaml (hunk 2 above)
Comment (iuriaranda) [reply rds-monitoring/values.yaml]: I've added the ReplicaLag metric.
These metrics are fetched "as is" from cloudwatch, so if an RDS instance doesn't generate a specific metric, it simply won't be fetched by the exporter.

**Comment by iuriaranda** [comment]
@duboisph @simonrondelez can you re-review? I've added some alerts as well and made some changes.

**Comment by iuriaranda** [comment]
Btw, I've deployed the chart in our cluster, in case you want to check out the alerts and metrics.

**Comment by iuriaranda** [comment]
Also, I'm a bit worried about how we monitor AWS resources with the cloudwatch exporter at the moment. Because like this, we're monitoring all RDS instances in the aws account, both production and staging, and this alerts will trigger regardless of the environment, and I see no way to fix that. Not sure if that's acceptable.

**Comment by duboisph** [comment]
@iuriaranda Yeah, valid concenr and a tricky one. Afaik for monitoring SQS it's only deployed to production for some customers (no alerts I think).

But to filter RDS here, you should be able to filter on a Dimension normally. Eg we do so for ElasticSearch too: https://github.com/skyscrapers/terraform-awselasticsearch/blob/master/elasticsearch_k8s_monitoring/templates/elasticsearch-monitoring-values.yaml.tpl#L30-L31. Unfortunately there it is per cluster, not ideal either (lucky just very limited metrics we request).

Maybe it's possible to filter out via AWS tags, so we can monitor eg per environment instead?

**Review by duboisph — DISMISSED** [review]
Apart from the above concerns, the PR itself looks good to me.

File: rds-monitoring/templates/prometheusrules.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            runbook_url: 'https://github.com/skyscrapers/documentation/tree/master/runbook.md#alert-name-rdsdiskqueuedepthhigh'
+        - alert: RDSCPUUsageHigh
+          expr: aws_rds_cpuutilization_average{job="{{ .Release.Name }}-prometheus-cloudwatch-exporter"} > 95
+          for: 10m
+          labels:
+            severity: info
+            group: persistence
+          annotations:
+            message: 'CPU usage for RDS instance {{`{{ $labels.dbinstance_identifier }}`}} is higher than 95%.'
+            runbook_url: 'https://github.com/skyscrapers/documentation/tree/master/runbook.md#alert-name-rdscpuusagehigh'
```
Comment (simonrondelez) [inline rds-monitoring/templates/prometheusrules.yaml:60]: I would add an alert for replicalag:
```
expr: aws_rds_replicalag_average{job="{{ .Release.Name }}-prometheus-cloudwatch-exporter"} > 30
```

Comment (iuriaranda) [reply rds-monitoring/templates/prometheusrules.yaml:60]: done

**Review by simonrondelez — DISMISSED** [review]

**Comment by iuriaranda** [comment]
Ok, after some digging I've found a couple of ways we could filter environment resources:

- we can configure the cloudwatch-exporter to export an "info" metric to Prometheus, one that combines the resource id with all its AWS tags. This way we can filter production resources in the prometheus alert expression.
- we can configure the cloudwatch-exporter to only export metrics for production resources via tag filter. This way just the production resource metrics would end up in Prometheus and we won't need to filter in the alert expression. Although I think it could still be useful to have non-production resource metrics in Prometheus in any case.

Cost-wise, there shouldn't be any change as the number of API calls will be the same afaik

**Comment by iuriaranda** [comment]
I've used the tag filtering option in cloudwatch exporter to only alert on metrics from production DBs. The metrics for all instances will still end up in Prometheus in any case, but they'll be filtered out in the alert expressions.

@duboisph and @simonrondelez can you review again?

**Review by duboisph — COMMENTED** [review]
Having the tag filter hardcoded in the alerts makes it not really dynamic. Ok if you use our terraform-rds module, but in any case tagging can be different.

Can you instead use a template for it so we can override?

Also need to fix the chart version conflict :P

**Comment by iuriaranda** [comment]
@duboisph I've made it configurable, and you can now also disable the filtering altogether.

**Review by duboisph — APPROVED** [review]

**Pull request merged**